doc/sphinx/docs/python/tutorials/coding_udi.rst

```diff
@@ -0,0 +1,81 @@
+on some pygmo classes. The above UDI can then be used to construct a :class:`~pygmo.island` (similarly to how UDP can be used to construct :class:`~pygmo.problem`, etc..).
+
+.. doctest::
+
+    >>> isl = pg.island(algo = pg.de(100), prob = pg.ackley(5), udi = my_isl(), size = 20)
+    >>> print(isl) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+    Island name: Its my island!
+        Status: idle
+    <BLANKLINE>
+    Algorithm: Differential Evolution
+    <BLANKLINE>
+    Problem: Ackley Function
+    <BLANKLINE>
+    Population size: 20
+        Champion decision vector: [...
+        Champion fitness: [...
+
+That was easy! Lets now understand what we actually did. The object ``isl`` now contains our UDI and, upon construction will open a thread and delegate the execution of
+``run_evolve(self, algo, pop)`` to it upon call to :func:`~pygmo.island.evolve()`. 
+
+But there is a catch. We are in python! So it is not possible, in general, to have the same interpreter execute instructions in parallel as,
+in most of the popular python language implementations, memory management is not thread-safe. So, while the code above is perfectly fine and will
+work with pygmo, a set of ``my_isl`` running evolutions will not run in parallel as each :class:`~pygmo.island`, when executing its :func:`~pygmo.island.evolve()` 
+acquires the GIL (Global Interpreter Lock) and holds it during the :func:`~pygmo.island.evolve()` execution. 
+
+As a consequence, the following code:
+
+    >>> archi = pg.archipelago(n = 5, algo = pg.de(100), prob = pg.rosenbrock(10), pop_size = 20, udi = my_isl())
+    >>> archi.evolve()
+
+will not run evolution in parallel (only using different threads).
+
+To code properly an UDI one need to code the ``def run_evolve(self, algo, pop)`` so that the GIL is released during the offload of the evolution task to a separate process.
+An example on how this can be achieved using, for example the multiprocessing module of python. Let us have a look at some code snippets from the  :class:`~pygmo.mp_island`
+
+.. doctest::
+
+   >>> def _evolve_func(algo, pop): # doctest : +SKIP
+   ...     return algo.evolve(pop)
+   >>> class mp_island(object): # doctest : +SKIP
+   ...     def __init__(self):
+   ...         # Init the process pool, if necessary.
+   ...         mp_island.init_pool()
+   ...
+   ...     def run_evolve(self, algo, pop):
+   ...         with mp_island._pool_lock:
+   ...             res = mp_island._pool.apply_async(_evolve_func, (algo, pop))
+   ...         return res.get()
+
+The full details are here not reported and can be read in the :class:`~pygmo.mp_island` code. In a nutshell, what happens is that the ``algo.evolve(pop)`` gets offloaded to
+a process (in a shared pool inited upon construction calling the ``init_pool`` static method). The instruction ``res.get()``, makes the thread where ``run_evolve``
```

Comment:
Maybe can put a link to the actual init_pool method? I think it's a documented bit of the public interface of mp_island.